simplemind/models.py

```diff
@@ -22,34 +23,41 @@ def __repr__(self):
         return str(self)
 
 
-class BasePlugin:
+class BasePlugin(SMBaseModel, ABC):
     """The base conversation plugin class."""
 
     # Plugin metadata.
     meta: Dict[str, Any] = {}
 
-    def initialize_hook(self, conversation: "Conversation"):
+    @abstractmethod
```

Comment:
please remove the @abstractmethods, as they require all of them to be implemented in the subclass, and someone may want only one

Reply:
fixed this